railties/lib/rails/commands/server.rb

```diff
@@ -94,7 +94,8 @@ def default_options
         environment:        (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || "development").dup,
         daemonize:          false,
         caching:            nil,
-        pid:                Options::DEFAULT_PID_PATH
+        pid:                Options::DEFAULT_PID_PATH,
+        restart_cmd:        restart_command
```

Comment:
I tested on Thin, Unicorn and Webrick. The `restart_cmd` has no effect on them. Meaning `rails restart` continues to **not work** while using those servers as before.